src/bundle/mutable.js

```diff
@@ -0,0 +1,143 @@
+   * code that is considered to have followed JS best practices, if this
+   * previous code used assignment to override.
+   *
+   * To work around this mistake, deepFreeze(), prior to freezing, replaces
+   * selected configurable own data properties with accessor properties which
+   * simulate what we should have specified -- that assignments to derived
+   * objects succeed if otherwise possible.
+   */
+  function beMutable(obj, prop, desc) {
+    if ('value' in desc && desc.configurable) {
+      const { value } = desc;
+
+      // eslint-disable-next-line no-inner-declarations
+      function getter() {
+        return value;
+      }
+
+      // Re-attach the data property on the object so
+      // it can be found by the deep-freeze traversal process.
+      getter.value = value;
+
+      // eslint-disable-next-line no-inner-declarations
+      function setter(newValue) {
+        if (obj === this) {
+          throw new TypeError(
+            `Cannot assign to read only property '${prop}' of object '${obj}'`,
+          );
+        }
+        if (hasOwnProperty.call(this, prop)) {
+          this[prop] = newValue;
+        } else {
+          defineProperty(this, prop, {
+            value: newValue,
+            writable: true,
+            enumerable: desc.enumerable,
+            configurable: desc.configurable,
+          });
+        }
+      }
+
+      defineProperty(obj, prop, {
+        get: getter,
+        set: setter,
+        enumerable: desc.enumerable,
+        configurable: desc.configurable,
+      });
+    }
+  }
+
+  function beMutableProperties(obj) {
+    if (!obj) {
+      return;
+    }
+    const descs = getOwnPropertyDescriptors(obj);
+    if (!descs) {
+      return;
+    }
+    ownKeys(obj).forEach(prop => beMutable(obj, prop, descs[prop]));
+  }
+
+  /**
+   * These properties are subject to the override mistake
+   * and must be converted before freezing.
+   */
+  function repairDataProperties(intrinsics) {
+    const { global: g, anonIntrinsics: a } = intrinsics;
+
+    const toBeRepaired = [
+      g.Object.prototype,
+      g.Array.prototype,
+      g.Boolean.prototype,
+      g.Date.prototype,
+      g.Number.prototype,
+      g.String.prototype,
+
+      g.Function.prototype,
+      a.GeneratorFunction.prototype,
+      a.AsyncFunction.prototype,
+      a.AsyncGeneratorFunction.prototype,
+
+      a.IteratorPrototype,
+      a.ArrayIteratorPrototype,
+
+      g.DataView.prototype,
+
+      a.TypedArray,
+      g.Int8Array.prototype,
+      g.Int16Array.prototype,
+      g.Int32Array.prototype,
+      g.Uint8Array,
+      g.Uint16Array,
+      g.Uint32Array,
+
+      g.Error.prototype,
+      g.EvalError.prototype,
+      g.RangeError.prototype,
+      g.ReferenceError.prototype,
+      g.SyntaxError.prototype,
+      g.TypeError.prototype,
+      g.URIError.prototype,
+    ];
+
+    // Promise may be removed from the whitelist
+    const PromisePrototype = g.Promise && g.Promise.prototype;
```

Comment:
What else can be de-whitelisted? everything?

Reply:
I don't remember why `Promise` is treated this way. The only smaller whitelist I know of is Jessie's, but it still include `Promise`. 

Attn @michaelfig Does this PR need to be adjusted to accommodate Jessie's smaller whitelist? What do you think about putting into the SES whitelist also the listing of what properties should be repaired, rather than using a separate `toBeRepaired` list?

Reply:
Added a comment that the `toBeRepaired` list should be provided externally